pkg/egresspolicy/egresspolicy.go

```diff
@@ -0,0 +1,520 @@
+		return false, err
+	}
+
+	m.policyConfigs[config.id] = &config
+	pods := m.getLocalPodsForPolicy(&config)
+	if len(pods) == 0 {
+		return true, nil
+	}
+	m.upsertPolicyPod(&config, pods...)
+	return true, nil
+}
+
+// Returns a slice of endpoint pods metadata that are selected by the given policy config.
+func (m *Manager) getLocalPodsForPolicy(config *EgressPolicyConfig) []*podMetadata {
+	var (
+		retPods []*podMetadata
+		podData *podMetadata
+		err     error
+	)
+
+	podStore := m.storeGetter.GetStore("pod")
+	for _, podItem := range podStore.List() {
+		pod, ok := podItem.(*slimcorev1.Pod)
+		if !ok {
+			continue
+		}
+		podIPs := k8sUtils.ValidIPs(pod.Status)
+		if len(podIPs) == 0 {
+			continue
+		}
+		if podData, err = m.getPodMetadata(pod, podIPs); err != nil {
+			log.WithError(err).WithFields(logrus.Fields{
+				logfields.K8sPodName:   pod.Name,
+				logfields.K8sNamespace: pod.Namespace,
+			}).Error("failed to get valid pod metadata")
+			continue
+		}
+		if k8sUtils.GetLatestPodReadiness(pod.Status) != slimcorev1.ConditionTrue {
+			continue
+		}
+		if !config.policyConfigSelectsPod(podData) {
+			continue
+		}
+		retPods = append(retPods, podData)
+	}
+
+	return retPods
+}
+
+// Deletes the internal state associated with the given policy, including egress ebpf map
+func (manager *Manager) DeleteEgressPolicy(config EgressPolicyConfig) error {
+	manager.mutex.Lock()
+	defer manager.mutex.Unlock()
+
+	storedConfig := manager.policyConfigs[config.id]
+	if storedConfig == nil {
+		return fmt.Errorf("local egress policy to be deleted not found")
+	}
+	log.WithFields(logrus.Fields{"policyID": config.id}).
+		Debug("Delete local egress policy")
+
+	for podId, policies := range manager.policyPods {
+		var newPolicyList []policyID
+		// make a new list excluding policy that is to be deleted
+		for _, policyId := range policies {
+			if policyId == storedConfig.id {
+				// found policy to pod mapping, need to delete egress map entry
+				// identified by pod and config
+				if podData, ok := manager.podDataStore[podId]; ok {
+					manager.deleteEgressMap(storedConfig, podData)
+				} else {
+					log.WithFields(logrus.Fields{"policyID": config.id}).
+						Warn("Did not find matching pod for policy deletion")
+				}
+			} else {
+				newPolicyList = append(newPolicyList, policyId)
+			}
+		}
+		if len(newPolicyList) > 0 {
+			manager.policyPods[podId] = newPolicyList
+		} else {
+			// emply policy list, do not delete podDataStore here since
+			// pod data remains unchanged
+			delete(manager.policyPods, podId)
+		}
+	}
+	delete(manager.policyConfigs, config.id)
+	return nil
+}
+
+func (manager *Manager) OnAddPod(pod *slimcorev1.Pod) {
+	manager.mutex.Lock()
+	defer manager.mutex.Unlock()
+
+	if len(manager.policyConfigs) == 0 {
+		return
+	}
+	// If the pod already exists in the internal cache, ignore all the subsequent
+	// events since they'll be handled in the OnUpdatePod callback.
+	// GH issue #13136
+	// TODO add unit test
```

Comment:
Please add a note in https://github.com/cilium/cilium/issues/13575 instead if you're not planning to address it in this pull request. Otherwise, we will most likely forget about it.

Reply:
Done. I will file a bug regarding all the items that is not covered in this PR but are expected to be done later.
Thanks!